src/GraphQL/Execution/Nodes/ArrayExecutionNode.cs

```diff
@@ -56,6 +59,38 @@ public override object ToValue()
             return items;
         }
 
+        public override bool ClearErrorNodes()
+        {
+            if (Items == null)
+                return true;
+
+            if (Items.Count == 0)
+                return false;
+
+            var ok = false;
+
+            for (int i = 0; i < Items.Count; ++i)
+            {
+                var item = Items[i];
+                bool valueIsNull = item.ClearErrorNodes();
+
+                if (valueIsNull && !ok)
+                {
+                    if (((ListGraphType)GraphType).ResolvedType is NonNullGraphType)
+                    {
+                        Items = null;
```

Comment:
I suspect changing `Items` and `SubFields` might cause problems. I'm not sure.

Reply:
It might change the effect of accessing certain properties of a copied IResolveFieldContext after execution...

Reply:
Or might not since maybe sub fields is cloned also...

Reply:
I would think about how to achieve a one-time setting of this property. It may be that this is not yet possible.

Reply:
I think this is a separate issue that requires attention. We need to stop execution as early as possible. Imagine `ArrayExecutionNode` of type `[Person!]` with 100000 child nodes. First element yields `null` in `SetArrayItemNodes`, so...

Reply:
Also what about `protected virtual void ValidateNodeResult(ExecutionContext context, ExecutionNode node)` ? I have a suspicion that all checks in `ClearNodeErrors` will never work...

Reply:
SetArrayItemNodes executes prior to the execution of child object nodes. Checking if the child is null is impossible in that case. 

ValidateNodeResult runs at its appropriate time. It is not supposed to throw errors for each parent node that is coerced to null because of a child execution error. 

The code executes the same as previously and the checks are essential. They were within ToValue, and I moved them out and into a method that you are required to run immediately prior to calling ToValue.

Reply:
(Serializing the node itself is essentially the same as calling ToValue)

Reply:
OK. I published another issue regarding this.